sql/sql_test: fix timeout in TestRaceWithBackfill

spilchen opened the pull request:
We have seen test flakes with the TestRaceWithBackfill timing out when run with race detection enabled. This will ensure the kv batch size is a reasonable value by forcing production values, overridding any metamorphic constants that may be in use.

Epic: None
Closes #130013
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130956)
<!-- Reviewable:end -->

rafiss reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130956)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/sql/schema_changer_test.go` line 18 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130956#-O75-gr960KitklUp5OS:-O75-gr960KitklUp5OT:b7evlvs) ([raw file](https://github.com/cockroachdb/cockroach/blob/35147ba1674752d9d3125268be6cf5c7061fbe8d/pkg/sql/schema_changer_test.go#L18)):*
> ```Go
> 	"database/sql/driver"
> 	"fmt"
> 	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
> ```

nit: i believe this import needs to be sorted to pass linting checks


<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130956#-:-O75-mSt-Gc6w6K1ORod:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen committed:
sql/sql_test: fix timeout in TestRaceWithBackfill

We have seen test flakes with the TestRaceWithBackfill timing out when
run with race detection enabled. This will ensure the kv batch size is a
reasonable value by forcing production values, overridding any
metamorphic constants that may be in use.

Epic: None
Closes #130013
Release note: None

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130956)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @fqazi)

___
*[`pkg/sql/schema_changer_test.go` line 18 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130956#-O75-gr960KitklUp5OS:-O75Uu3Q3LdnmYT0siiT:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/35147ba1674752d9d3125268be6cf5c7061fbe8d/pkg/sql/schema_changer_test.go#L18)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: i believe this import needs to be sorted to pass linting checks
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

spilchen commented:
I might have to put the merge on hold. It seems the new version of the testcase flaked when run under stress. I see this:
```
Failed
=== RUN   TestRaceWithBackfill
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestRaceWithBackfill3985636730
    test_log_scope.go:81: use -show-logs to present logs inline
    schema_changer_test.go:344: schema change ALTER TABLE t.test ADD COLUMN x DECIMAL DEFAULT (DECIMAL '1.4') CHECK (x >= 0) took 1m18.83148533s
    schema_changer_test.go:344: schema change ALTER TABLE t.test DROP pi took 2m20.374781709s
    schema_changer_test.go:344: schema change CREATE UNIQUE INDEX foo ON t.test (v) took 2m38.738254644s
    schema_changer_test.go:344: schema change CREATE INDEX bar ON t.test(k) STORING (v) took 6m55.718680694s
    schema_changer_test.go:430: pq: batch timestamp 1726687441.033651964,2 must be after replica GC threshold 1726687474.999377272,0
    schema_changer_test.go:734: e = 201, v = 204
    schema_changer_test.go:745: read the wrong number of rows: e = 201, v = 202
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestRaceWithBackfill3985636730
--- FAIL: TestRaceWithBackfill (1137.84s)
```

spilchen commented:
I’m proceeding with the merge. The issue I noticed during the stress run is the same as #131401 and will be addressed separately.

bors r=fqazi,rafiss
